sympy/integrals/manualintegrate.py

```diff
@@ -1553,9 +1588,32 @@ def eval_reciprocal_sqrt_quadratic(a, b, c, integrand, x):
     return log(2*sqrt(c)*sqrt(a+b*x+c*x**2)+b+2*c*x)/sqrt(c)
 
 
+@evaluates(SqrtQuadraticDenomRule)
+def eval_sqrt_quadratic_denom(a, b, c, coeffs: list[Expr], integrand, x):
+    R"""
+    Integrate poly/sqrt(a+b*x+c*x**2) using the recursion formula below. coeffs are coefficients of the polynomial.
+    Let $I_n=\int\frac{x^n}{\sqrt{a+bx+cx^2}}\mathrm{d}x$,
+    then $I_n=\frac{1}{cn}x^{n-1}\sqrt{a+bx+cx^2}-\frac{2n-1}{2n}\frac{b}{c}I_{n-1}-\frac{n-1}{n}\frac{a}{c}I_{n-2}$
+    """
+    result_coeffs = []
+    coeffs = coeffs.copy()
+    for i in range(len(coeffs)-2):
+        n = len(coeffs)-1-i
+        coeff = coeffs[i]/(c*n)
+        result_coeffs.append(coeff)
+        coeffs[i+1] -= (2*n-1)*b/2*coeff
+        coeffs[i+2] -= (n-1)*a*coeff
+    d, e = coeffs[-1], coeffs[-2]
+    s = sqrt(a+b*x+c*x**2)
+    constant = d-b*e/(2*c)
+    I0 = 0 if constant == 0 else constant*_manualintegrate(inverse_trig_rule(IntegralInfo(1/s, x), degenerate=False))
+    return Add(*(result_coeffs[i]*x**(len(coeffs)-2-i) for i in range(len(result_coeffs))), e/c)*s + I0
```

Comment:
I would rewrite this as something like:
```python
    if constant == 0:
        I0 = 0
    else:
        tr = inverse_trig_rule(IntegralInfo(1/s, x))
        I0 = constant * _manualintegrate(tr, degenerate=False))
    N = len(result_coeffs)
    terms = (result_coeffs[i] * x**(N-2-i) for i in range(N))
    return I0 + s * Add(*terms, e/c)
```
What does `degenerate=False` do?

Reply:
`degenerate=False` tells the function not to consider degenerate cases. It’s set `False` only when it’s called recursively, where the caller already assumes `c` nonzero.
It’s here to avoid nested `Piecewise`, where outer already has `Ne(c, 0)` but inner still asks `Ne(c, 0)` or not.
I’ll put it into docstring.